File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    answer = 0
    
    def dfs(index, sum):
        nonlocal answer
        # 끝까지 갔는 지 확인
        if index >= len(numbers):
            if sum == target:
                answer += 1
            return
        # 타켓과 같은 지 확인
        
        dfs(index + 1 , sum + numbers[index])
        
        dfs(index + 1 , sum - numbers[index])
        
    dfs(0, 0)
    
    return answer

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    int answer = 0;
    public int solution(int[] numbers, int target) {
        
        
        // dfs
        
        dfs(0, 0, numbers, target);
        
        return answer;
    }
    
    public void dfs(int index, int current, int[] numbers, int target) {
            // exception - 도달하면
            if ( index == numbers.length ){
                if ( current == target) answer += 1;
                return;
            } 
            
            // 빼기, 더하기
            dfs(index + 1 , current + numbers[index], numbers, target);
            dfs(index + 1 , current - numbers[index], numbers, target);

        }
}